packages/core/src/nuxt.js

```diff
@@ -26,7 +26,7 @@ export default class Nuxt extends Hookable {
     // Deprecated hooks
     this.deprecateHooks({
       'render:context': 'render:routeContext',
-      'render:routeContext': 'vue-renderer:afterRender',
+      'render:routeContext': 'vue-renderer:ssr:context',
```

Comment:
These hooks are not identical by parameters! `render:routeContext` accepts `renderContext.nuxt` while `vue-renderer:ssr:context` accepts `renderContext` ([src](https://github.com/nuxt/nuxt.js/blob/dev/packages/vue-renderer/src/renderers/ssr.js) 

Ref in v2.4.0: https://github.com/nuxt/nuxt.js/blob/v2.4.0/packages/vue-renderer/src/renderer.js#L373

Reply:
But I don't see there is a `vue-renderer:afterRender`

Reply:
Indeed. This is why it is kept for next major: https://github.com/nuxt/nuxt.js/blob/dev/packages/vue-renderer/src/renderers/ssr.js#L94

Reply:
I mean the error message is `render:routeContext hook has been deprecated, please use vue-renderer:afterRender`, but there is no `vue-renderer:afterRender`, users who migrate to new hook won't get hook worked.

Reply:
If only diff is param, we need to update doc.

Reply:
There are 3 problems:
- `vue-renderer:afterRender` does not exists
- Recursive deprecate not being supported
- deprecated warning is not customizable (for example to mention param changes)

Working on it will make a PR :)

Reply:
@pi0 regarding params indeed it accepts different param. I have mentioned that in the [docs pull request](https://github.com/nuxt/docs/pull/1832/files). Since this is merged it makes sense to merge that pull request too.